gsw/tests/test_check_functions.py

```diff
@@ -45,7 +45,7 @@
 mfuncnames = [mf.name for mf in mfuncs]
 
 
-@pytest.fixture(params=[-360, 0, 360])
+@pytest.fixture(params=[-360., 0., 360.])
```

Comment:
The hardest part was setting up the np2.0 test! This was the only failure we had due to https://numpy.org/devdocs/numpy_2_0_migration_guide.html#changes-to-numpy-data-type-promotion

Reply:
@efiring this is the only code change in this PR. Everything else is to be able to set a testing matrix with numpy 2.0. This PR also adds a test for the code generation step that serves both as test and developer breadcrumbs in case we need to do that locally.